pkg/k8sutil/k8sutil_test.go

```diff
@@ -533,33 +504,21 @@ func TestCreateOrUpdateImmutableFields(t *testing.T) {
 			Status: corev1.ServiceStatus{},
 		}
 
-		if err := CreateOrUpdateService(context.TODO(), svcClient, modifiedSvc); err != nil {
-			t.Fatal(err)
-		}
+		require.NoError(t, CreateOrUpdateService(context.TODO(), svcClient, modifiedSvc))
 
-		if !reflect.DeepEqual(service.Spec.IPFamilies, modifiedSvc.Spec.IPFamilies) {
-			t.Fatalf("services Spec.IPFamilies are not equal, expected %q, got %q",
-				service.Spec.IPFamilies, modifiedSvc.Spec.IPFamilies)
-		}
+		require.Equal(t, service.Spec.IPFamilies, modifiedSvc.Spec.IPFamilies, "services Spec.IPFamilies are not equal, expected %q, got %q",
+			service.Spec.IPFamilies, modifiedSvc.Spec.IPFamilies)
 
-		if !reflect.DeepEqual(service.Spec.ClusterIP, modifiedSvc.Spec.ClusterIP) {
-			t.Fatalf("services Spec.ClusterIP are not equal, expected %q, got %q",
-				service.Spec.ClusterIP, modifiedSvc.Spec.ClusterIP)
-		}
+		require.Equal(t, service.Spec.ClusterIP, modifiedSvc.Spec.ClusterIP, "services Spec.ClusterIP are not equal, expected %q, got %q",
+			service.Spec.ClusterIP, modifiedSvc.Spec.ClusterIP)
 
-		if !reflect.DeepEqual(service.Spec.ClusterIPs, modifiedSvc.Spec.ClusterIPs) {
-			t.Fatalf("services Spec.ClusterIPs are not equal, expected %q, got %q",
-				service.Spec.ClusterIPs, modifiedSvc.Spec.ClusterIPs)
-		}
+		require.Equal(t, service.Spec.ClusterIPs, modifiedSvc.Spec.ClusterIPs, "services Spec.ClusterIPs are not equal, expected %q, got %q",
+			service.Spec.ClusterIPs, modifiedSvc.Spec.ClusterIPs)
 
-		if !reflect.DeepEqual(service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy) {
-			t.Fatalf("services Spec.IPFamilyPolicy are not equal, expected %v, got %v",
-				service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy)
-		}
+		require.Equal(t, service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy, "services Spec.IPFamilyPolicy are not equal, expected %v, got %v",
+			service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy)
 
-		if !reflect.DeepEqual(service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy) {
-			t.Fatalf("services Spec.IPFamilyPolicy are not equal, expected %v, got %v",
-				service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy)
-		}
+		require.Equal(t, service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy, "services Spec.IPFamilyPolicy are not equal, expected %v, got %v",
+			service.Spec.IPFamilyPolicy, modifiedSvc.Spec.IPFamilyPolicy)
```

Comment:
This block of code is repeated. I also replaced it twice but was this neeeded?

Reply:
Oh nice catch, let's delete the duplicated assertion!